sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -109,5 +138,20 @@ class BulkWriter(container: CosmosAsyncContainer,
   private def cancelWork(): Unit = {
     subscriptionDisposable.dispose()
   }
+
+  private def shouldIgnore(exception: Exception) : Boolean = {
+    // ignore 409 on create-item
+    writeConfig.itemWriteStrategy == ItemWriteStrategy.ItemAppend &&
+      exception.isInstanceOf[CosmosException] &&
```

Comment:
We are casting to CosmosExceptions often, can't we instead use pattern matching? In the calling method, if the exception is not CosmosException would would not even call shouldIgnore or shouldRetry.

Something like

```
Option(resp.getException) match 
case Some(cosmosException: CosmosException) if !shouldIgnore(cosmosException) && shouldRetry(cosmosException) => 
     // requeue
case _ => 
    captureIfFirstFailure(resp.getException)
    cancelWork()
```

Then the shouldRetry and shouldIgnore can take CosmosException and would reduce the `isInstanceOf` or `asInstanceOf` usage.

Reply:
Also helps remove the `null` comparisons. Just an idea

Reply:
good suggestion @ealsur thank you. addressed.